mycli/main.py

```diff
@@ -811,16 +811,29 @@ def format_output(self, title, cur, headers, expanded=False,
             output.append(title)
 
         if cur:
-            rows = list(cur)
+            column_types = None
+            if hasattr(cur, 'description'):
+                def sanitize(col):
+                    if col is converters.through:
+                        return text_type
+                    else:
+                        return col
+                column_types = [sanitize(converters.decoders[col[1]])
+                                for col in cur.description]
+
+            if max_width is not None:
+                cur = list(cur)
+
             formatted = self.formatter.format_output(
-                rows, headers, format_name='vertical' if expanded else None,
+                cur, headers, format_name='vertical' if expanded else None,
+                column_types=column_types,
```

Comment:
Should we set column_types in `output_kwargs` (i.e. `output_kwargs['column_types'] = ...`) so that this call to `format_output()` *and* the call below will both get the column types?